core/templates/pages/exploration-player-page/layout-directives/content-language-selector.component.spec.ts

```diff
@@ -117,16 +117,17 @@ describe('Content language selector component', () => {
     fixture.detectChanges();
   }));
 
-  it('should correctly initialize selectedLanguageCode and ' +
-     'languagesInExploration', () => {
+  it('should correctly initialize selectedLanguageCode, ' +
+     'newLanguageCode and languagesInExploration', () => {
     expect(component.selectedLanguageCode).toBe('fr');
+    expect(component.newLanguageCode).toBe('fr');
     expect(component.languageOptions).toEqual([
       {value: 'fr', displayed: 'français (French)'},
       {value: 'zh', displayed: '中文 (Chinese)'},
       {value: 'en', displayed: 'English'}
     ]);
   });
```

Comment:
```suggestion
  it('should correctly initialize selectedLanguageCode and , ' +
     'languagesInExploration', () => {
    expect(component.selectedLanguageCode).toBe('fr');
    expect(component.languageOptions).toEqual([
      {value: 'fr', displayed: 'français (French)'},
      {value: 'zh', displayed: '中文 (Chinese)'},
      {value: 'en', displayed: 'English'}
    ]);
  });

  it('should correcly initialize newLanguageCode', () => {
    let mockWindowRef = new MockWindowRef();
    let url = new URL(mockWindowRef.nativeWindow.location.href);

    expect(url.searchParams.get('initialContentLanguageCode')).toBe('fr');

    component.ngOnInit();

    expect(component.newLanguageCode).toBe('fr');
  });

```

Reply:
also add this class after line numb.70
class MockWindowRef {
  nativeWindow = {
    location: {
      href: 'http://localhost:8181/explore/wZiXFx1iV5bz?initialContentLanguageCode=fr'
    }
  };
}
this is simply mocking the WindowRef with it required properties

Reply:
Thanks @Shivkant-Chauhan for the help.